src/main/java/org/eclipse/golo/compiler/ir/GoloFunction.java

```diff
@@ -334,4 +335,30 @@ public void walk(GoloIrVisitor visitor) {
   protected void replaceElement(GoloElement original, GoloElement newElement) {
     throw cantReplace();
   }
+
+  @Override
+  public int hashCode() {
+    int hash = 5;
+    hash = 89 * hash + Objects.hashCode(this.name);
+    hash = 89 * hash + Objects.hashCode(this.getArity());
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) {
+      return false;
+    }
+    if (getClass() != obj.getClass()) {
+      return false;
+    }
+    final GoloFunction other = (GoloFunction) obj;
+    if (!Objects.equals(this.name, other.name)) {
+      return false;
+    }
+    if (this.getArity() != other.getArity()) {
+      return false;
+    }
+    return true;
+  }
```

Comment:
Yes but I mean after the cast, you can just return the boolean expression instead of the two `if`s (since `&&` is lazy)